### ISSUE TITLE
pip install not working on aarch64

### ISSUE THREAD

**Issue by PinkJohnOfUs** [eclipse-zenoh/zenoh-python#41]
Trying to install on nvidia jetson:
```
(venv) k@os:$ uname -i
aarch64
(venv) k@os:$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.6 LTS
Release:        18.04
Codename:       bionic
(venv) k@os:$ python --version
Python 3.6.9
```
But fails with:
```
$ pip install eclipse-zenoh
Collecting eclipse-zenoh
  Downloading eclipse_zenoh-0.5.0_b8.tar.gz (74 kB)
     |████████████████████████████████| 74 kB 846 kB/s            
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing wheel metadata (pyproject.toml) ... done
Building wheels for collected packages: eclipse-zenoh
  Building wheel for eclipse-zenoh (pyproject.toml) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/alf/proj/zenoh_play/venv/bin/python3 /home/alf/proj/zenoh_play/venv/lib/python3.6/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /tmp/tmp0rv6g8c0
       cwd: /tmp/pip-install-zineoisv/eclipse-zenoh_8413ab83c85a404597df68fa8f7150a8
  Complete output (255 lines):
  Running `maturin pep517 build-wheel -i /home/alf/proj/zenoh_play/venv/bin/python3`
  ⚠️  Warning: Please use maturin in pyproject.toml with a version constraint, e.g. `requires = ["maturin>=0.11,<0.12"]`. This will become an error.
TLDR;
  error: could not compile `shared_memory`
  
  To learn more, run the command again with --verbose.
  warning: build failed, waiting for other jobs to finish...
  error: build failed
  💥 maturin failed
    Caused by: Failed to build a native library through cargo
    Caused by: Cargo build finished with "exit status: 101": `cargo rustc --message-format json --manifest-path Cargo.toml --release --lib --`
  🔗 Found pyo3 bindings with abi3 support for Python ≥ 3.6
  🐍 Not using a specific python interpreter (With abi3, an interpreter is only required on windows)
  error[E0164]: expected tuple struct or tuple variant, found associated function `nix::Error::Sys`
    --> /home/alf/.cargo/registry/src/github.com-1ecc6299db9ec823/shared_memory-0.11.4/src/unix.rs:87:13
     |
  87 |         Err(nix::Error::Sys(Errno::EEXIST)) => return Err(ShmemError::MappingIdExists),
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `fn` calls are not allowed in patterns
     |
     = help: for more information, visit https://doc.rust-lang.org/book/ch18-00-patterns.html
  
  error[E0164]: expected tuple struct or tuple variant, found associated function `nix::Error::Sys`
    --> /home/alf/.cargo/registry/src/github.com-1ecc6299db9ec823/shared_memory-0.11.4/src/unix.rs:88:13
     |
  88 |         Err(nix::Error::Sys(e)) => return Err(ShmemError::MapCreateFailed(e as u32)),
     |             ^^^^^^^^^^^^^^^^^^ `fn` calls are not allowed in patterns
     |
     = help: for more information, visit https://doc.rust-lang.org/book/ch18-00-patterns.html
  
  
  error[E0164]: expected tuple struct or tuple variant, found associated function `nix::Error::Sys`
     --> /home/alf/.cargo/registry/src/github.com-1ecc6299db9ec823/shared_memory-0.11.4/src/unix.rs:103:13
      |
  103 |         Err(nix::Error::Sys(e)) => return Err(ShmemError::UnknownOsError(e as u32)),
      |             ^^^^^^^^^^^^^^^^^^ `fn` calls are not allowed in patterns
      |
      = help: for more information, visit https://doc.rust-lang.org/book/ch18-00-patterns.html
  
  
  error[E0164]: expected tuple struct or tuple variant, found associated function `nix::Error::Sys`
     --> /home/alf/.cargo/registry/src/github.com-1ecc6299db9ec823/shared_memory-0.11.4/src/unix.rs:119:13
      |
  119 |         Err(nix::Error::Sys(e)) => return Err(ShmemError::MapCreateFailed(e as u32)),
      |             ^^^^^^^^^^^^^^^^^^ `fn` calls are not allowed in patterns
      |
      = help: for more information, visit https://doc.rust-lang.org/book/ch18-00-patterns.html
  
  
  error[E0164]: expected tuple struct or tuple variant, found associated function `nix::Error::Sys`
     --> /home/alf/.cargo/registry/src/github.com-1ecc6299db9ec823/shared_memory-0.11.4/src/unix.rs:135:13
      |
  135 |         Err(nix::Error::Sys(e)) => return Err(ShmemError::MapOpenFailed(e as u32)),
      |             ^^^^^^^^^^^^^^^^^^ `fn` calls are not allowed in patterns
      |
      = help: for more information, visit https://doc.rust-lang.org/book/ch18-00-patterns.html
  
  
  error[E0164]: expected tuple struct or tuple variant, found associated function `nix::Error::Sys`
     --> /home/alf/.cargo/registry/src/github.com-1ecc6299db9ec823/shared_memory-0.11.4/src/unix.rs:150:13
      |
  150 |         Err(nix::Error::Sys(e)) => return Err(ShmemError::MapOpenFailed(e as u32)),
      |             ^^^^^^^^^^^^^^^^^^ `fn` calls are not allowed in patterns
      |
      = help: for more information, visit https://doc.rust-lang.org/book/ch18-00-patterns.html
  
  
  error[E0164]: expected tuple struct or tuple variant, found associated function `nix::Error::Sys`
     --> /home/alf/.cargo/registry/src/github.com-1ecc6299db9ec823/shared_memory-0.11.4/src/unix.rs:166:13
      |
  166 |         Err(nix::Error::Sys(e)) => return Err(ShmemError::MapOpenFailed(e as u32)),
      |             ^^^^^^^^^^^^^^^^^^ `fn` calls are not allowed in patterns
      |
      = help: for more information, visit https://doc.rust-lang.org/book/ch18-00-patterns.html
  
  
  error: aborting due to 7 previous errors
  
  
  For more information about this error, try `rustc --explain E0164`.
  
  Error: command ['maturin', 'pep517', 'build-wheel', '-i', '/home/alf/proj/zenoh_play/venv/bin/python3'] returned non-zero exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for eclipse-zenoh
Failed to build eclipse-zenoh
ERROR: Could not build wheels for eclipse-zenoh, which is required to install pyproject.toml-based projects
```

**Comment by JEnoch** [comment]
Hi @PinkJohnOfUs ,

The issue is the same than #39.
The root cause is a breaking change in a dependency (nix:0.22.0) that makes another required dependency (shared-memory:0.11.4) to fail building. Unfortunately, in eclipse_zenoh-0.5.0_b8.tar.gz we didn't add the Cargo.lock that would have fixed all dependencies versions.

Waiting for a new release of zenoh-python (I'm working on it with a new CI to build binary wheels also for aarch64), you need to build and install the version from `master` branch. See the instructions:
https://github.com/eclipse-zenoh/zenoh-python#how-to-build-it

If you also want to use the zenoh router, you'll need to use its `master` version, either [buidling it](https://github.com/eclipse-zenoh/zenoh#how-to-build-it), either using the Docker image `eclipse/zenoh:master`.

**Comment by PinkJohnOfUs** [comment]
Thank you for the response. I am able to build with maturin. But it does not solve my issue, because I need this zenoh-python as pip package within my ansible deployment and I did not manage to build it with from within ansible. So that is not the problem of this issue or your problem, but is it possible to leave this issue open until binary wheels are available for aarch64, please? ... then I could solve my ansible issue, when this issue is closed  :)

**Comment by Mallets** [comment]
This problem should have been fixed by a PR upstream to shared memory [https://github.com/elast0ny/shared_memory/pull/68](https://github.com/elast0ny/shared_memory/pull/68).

Closing the issue.
Please reopen it if the problem persists.
